lightkurve/correctors/pldcorrector.py

```diff
@@ -82,199 +116,509 @@ class PLDCorrector(object):
+            If `True`, all parameters will be optimized separately before
+            attempting to optimize all parameters together.  This will be
+            significantly slower but increases the likelihood of success.
+        **kwargs : dict
+            Dictionary of arguments to be passed to
+            `~lightkurve.correctors.PLDCorrector.create_pymc_model`.
 
+        Returns
+        -------
+        solution : dict
+            Maximum likelihood values.
+        """
+        if model is None:
+            model = self.create_pymc_model(**kwargs)
+        if start is None:
+            start = model.test_point
+
+        with model:
+            solution = xo.optimize(start=start, vars=[model.logs2])
+            # Optimizing parameters separately appears to make finding a solution more likely
+            if robust:
+                solution = xo.optimize(start=solution, vars=[model.logsigma])
+                solution = xo.optimize(start=solution, vars=[model.logrho, model.logsigma])
+                solution = xo.optimize(start=solution, vars=[model.logs2])
+            solution = xo.optimize(start=solution)  # Optimize all parameters
+
+        self.most_recent_solution_or_trace = solution
+        return solution
+
+    def sample(self, model=None, start=None, draws=1000, chains=4):
+        """Sample the systematics correction model.
+
+        Parameters
+        ----------
+        model : `pymc3.model.Model`
+            A pymc3 model.
+        start : dict
+            Initial parameter values to initiate the sampling. If `None`,
+            the output of this object's `optimize()` method will be used.
+        draws : int
+            Number of MCMC samples.
+        chains : int
+            Number of MCMC chains.
+
+        Returns
+        -------
+        trace : `~pymc3.backends.base.MultiTrace`
+            Trace object containing parameters and their samples.
+        """
+        # Create the model
+        if model is None:
+            model = self.create_pymc_model()
+        if start is None:
+            start = self.optimize(model=model)
+
+        # Initialize the sampler
+        sampler = xo.PyMC3Sampler()
+        with model:
+            # Burn in the sampler
+            sampler.tune(tune=np.max([int(draws*0.3), 150]),
+                         start=start,
+                         step_kwargs=dict(target_accept=0.9),
+                         chains=chains)
+            # Sample the parameters
+            trace = sampler.sample(draws=draws, chains=chains)
+
+        self.most_recent_solution_or_trace = trace
+        return trace
+
+    def correct(self, sample=False, remove_gp_trend=False, **kwargs):
+        """Returns a systematics-corrected light curve.
+
+        Parameters
+        ----------
+        sample : boolean
+            `True` will sample the output of the optimization
+            step and include robust errors on the output light curve.
+        remove_gp_trend : boolean
```

Comment:
Let's make sure these keywords have easy to understand names, that are consistent between correctors. Since we will also be having a GP corrector, could this be `remove_long_term_trends` or similar? Don't want people to get confused.